src/main/java/org/skriptlang/skript/bukkit/itemcomponents/equippable/elements/ExprEquipCompCameraOverlay.java

```diff
@@ -0,0 +1,106 @@
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.ValidationResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.itemcomponents.equippable.EquippableExperiment;
+import org.skriptlang.skript.bukkit.itemcomponents.equippable.EquippableWrapper;
+import org.skriptlang.skript.log.runtime.SyntaxRuntimeErrorProducer;
+
+@Name("Equippable Component - Camera Overlay")
+@Description({
+	"The camera overlay for the player when the item is equipped.",
+	"Example: The jack-o'-lantern view when having a jack-o'-lantern equipped as a helmet.",
+	"The camera overlay is represented as a namespaced key.",
+	"A namespaced key can be formatted as 'namespace:id' or 'id'; "
+		+ "Can only contain one ':' to separate the namespace and the id, alphanumeric characters, periods, underscores, and dashes.",
+	"Note that equippable component elements are experimental making them subject to change and may not work as intended."
+})
+@Example("set the camera overlay of {_item} to \"custom_overlay\"")
+@Example("""
+	set {_component} to the equippable component of {_item}
+	set the camera overlay of {_component} to "custom_overlay"
+	""")
+@RequiredPlugins("Minecraft 1.21.2+")
+@Since("INSERT VERSION")
+public class ExprEquipCompCameraOverlay extends PropertyExpression<EquippableWrapper, String> implements EquippableExperiment, SyntaxRuntimeErrorProducer {
+
+	static {
+		register(ExprEquipCompCameraOverlay.class, String.class, "camera overlay", "equippablecomponents");
+	}
+
+	private Node node;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		//noinspection unchecked
+		setExpr((Expression<EquippableWrapper>) exprs[0]);
+		node = getParser().getNode();
+		return true;
+	}
+
+	@Override
+	protected String @Nullable [] get(Event event, EquippableWrapper[] source) {
+		return get(source, wrapper -> {
+			NamespacedKey key = wrapper.getComponent().getCameraOverlay();
+			return key == null ? null : key.toString();
+		});
+	}
+
+	@Override
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
+			return CollectionUtils.array(String.class);
+		return null;
+	}
+
+	@Override
+	public void change(Event event, Object @Nullable [] delta, ChangeMode mode) {
+		NamespacedKey key = null;
+		if (delta != null && delta[0] instanceof String string) {
+			ValidationResult<NamespacedKey> validationResult = NamespacedUtils.checkValidation(string);
+			String validationMessage = validationResult.message();
+			if (!validationResult.valid()) {
+				error(validationMessage + " " + NamespacedUtils.NAMEDSPACED_FORMAT_MESSAGE);
```

Comment:
This formatting seems like it could be weird. Maybe this might work better:
```suggestion
				error(validationMessage + ". " + NamespacedUtils.NAMEDSPACED_FORMAT_MESSAGE);
```
Or, consider appending this in NamespacedUtils where the message is constructed